### PR TITLE
The classes queue and exam are already to be reviewed

### PR DESCRIPTION
**Pull request by chosky**

### CODE REVIEW

File: evaluator/include/exam.h [hunk 1]
```diff
@@ -12,10 +12,35 @@ class Exam {
   int quantity;
 
  public:
-  Exam();
+  /**
+   * Constructor of Exam class
```
Comment (hamax97) [inline evaluator/include/exam.h]: Add . to the end of a comment. Also, saying that it is the Exam's class constructor is redundant, I would say _Main constructor._

File: evaluator/include/exam.h [hunk 2]
```diff
@@ -12,10 +12,35 @@ class Exam {
...
+
+  /**
+   * Destructor of Exam class
+   */
   ~Exam();
+
+  /**
+   * get the id of the exam
+   */
   int get_id();
```
Comment (hamax97) [inline evaluator/include/exam.h]: Same as above `get_exam_id`.

File: evaluator/include/queue.h [hunk 3]
```diff
@@ -16,10 +16,29 @@ class Queue {
   std::queue<Exam> queue;
 
  public:
-  Queue();
+  /**
+   * Constructor of Queue class
```
Comment (hamax97) [inline evaluator/include/queue.h]: Same as above, dot and redundancy.

File: evaluator/include/queue.h [hunk 4]
```diff
@@ -40,9 +66,16 @@ class InputQueue : public Queue {
 */
 class EvaluatorQueue : public Queue {
  private:
-  char type;
+  char _type;
```
Comment (hamax97) [inline evaluator/include/queue.h]: Why the underscore?.

File: evaluator/src/exam.cpp [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  this->queue_id = queue_id;
+  this->type = type;
+  this->quantity = quantity;
+}
+
+Exam::~Exam() { }
+
+int
+Exam::get_id() {
+  return id;
```
Comment (hamax97) [inline evaluator/src/exam.cpp]: Missing `this->`. Also in the rest of the functions.

File: evaluator/src/queue.cpp [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  for(int i = 0; i < this->queue.size(); ++i) {
+    Exam temporal_exam = queue.front();
+    exams.push_back(temporal_exam);
+    queue.pop();
+    queue.push(temporal_exam);
+  }
+  return exams;
+}
+
+InputQueue::InputQueue(int queue_id, QueueType type, int size) : Queue(QueueType type, int size){
```
Comment (hamax97) [inline evaluator/src/queue.cpp]: I think that the proper way of using the list of initialization is `: Queue(type, size)`. Are you sure of this?.

File: evaluator/src/queue.cpp [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  this->queue_id = queue_id;
+}
+
+InputQueue::~InputQueue() { }
+
+EvaluatorQueue::EvaluatorQueue(char _type, QueueType type, int size) {
```
Comment (hamax97) [inline evaluator/src/queue.cpp]: Missing father initialization.

File: evaluator/include/exam.h [hunk 8]
```diff
@@ -12,10 +12,35 @@ class Exam {
   int quantity;
 
  public:
-  Exam();
+  /**
+   * Constructor of Exam class
+   */
+  Exam(int id, int queue_id, ExamType type, int quantity);
```
[the remark this answers is not in the document]
Comment (chosky) [reply evaluator/include/exam.h]: Ok

File: evaluator/include/exam.h (hunk 1 above)
Comment (chosky) [reply evaluator/include/exam.h]: Ok

File: evaluator/include/exam.h (hunk 2 above)
Comment (chosky) [reply evaluator/include/exam.h]: Ok

File: evaluator/include/queue.h (hunk 3 above)
Comment (chosky) [reply evaluator/include/queue.h]: Ok

File: evaluator/include/queue.h (hunk 4 above)
Comment (chosky) [reply evaluator/include/queue.h]: To not have problems with the Queue type

File: evaluator/src/queue.cpp (hunk 6 above)
Comment (chosky) [reply evaluator/src/queue.cpp]: No, you have all the reason

File: evaluator/src/queue.cpp (hunk 7 above)
Comment (chosky) [reply evaluator/src/queue.cpp]: Ok

File: evaluator/include/queue.h (hunk 4 above)
Comment (hamax97) [reply evaluator/include/queue.h]: You're right. What about `reactive_queue_type`?, with this identifier we also avoid the confusion of sometimes using underscore and sometimes not.

File: evaluator/src/exam.cpp (hunk 5 above)
Comment (hamax97) [reply evaluator/src/exam.cpp]: ??

Comment (chosky) [reply evaluator/src/exam.cpp]: ok

File: evaluator/include/queue.h (hunk 4 above)
Comment (chosky) [reply evaluator/include/queue.h]: Ok, I will put that name

**Pull request merged**